### ISSUE TITLE
Move node representation onto state

### ISSUE THREAD

**Issue by esjewett** [humanitiesplusdesign/fibra#41]

**Comment by esjewett** [comment]
Per Eetu, we store the following on the client-side (in the state):

* ids of nodes visible in construct (the "construct set")
* position of nodes in construct

**Comment by jiemakel** [comment]
We could (and for demo purposes should) also support storing the client side state on the server. This would be loaded in at startup, and serialized back specifically based on user action ("Save/Share project/canvas/set"). This isn't a priority, however.

**Comment by esjewett** [comment]
Yup, agreed. Once something is on the state, then it's very straightforward to serialize and we don't have to go looking for all the little pieces we want to save. Currently it's just JSON but I'm going to switch it over to immutable.js relatively soon. Both can be serialized.
